src/driver/platform_rpi.cpp

```diff
@@ -0,0 +1,140 @@
+
+/****************************************************************
+ * platform_rpi.cpp   
+ * HSCDTD008A Library Extention for RPI
+ * Tilman Glötzner
+ * Original Creation Date: 2022-09-07
+ *
+ * Distributed as-is; no warranty is given.
+ ***************************************************************/
+
+
+#include "platform.h"
+#include "hscdtd008a_driver.h"
+#include <stdio.h>
+#include <sys/ioctl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <unistd.h>
+#include <string.h>
+#include <linux/i2c-dev.h>
+#define I2C_ADDR 0x0f
+#define device "/dev/i2c-1"
+#ifndef I2C_M_RD
+#include <linux/i2c.h>
+#endif
```

Comment:
I am uncertain if this functionality is related. If it is not, please separate defines from includes (even conditional includes)

Reply:
These are the definitions need to access the ii2c driver.  I2C_ADDR is the I2C address of the the HSCDTD008A.  'device'  declares the i2c bus to be used (raspberry pi has 1 or 2 buses depending on the model).

The linux/raspberry/Posix driver model is different from that of the  arduino:
open returns a file handle, and all subsequent functions such as read and write use the file handle. The beauty of this approach is that with open, close, ioctl, read, and write practically everything can be abstracted. A graphics card can be accessed with the same functions as a disk drive, a file system, s real time clock or an usb device  -- even though the driver below this interface is radically different. And in case there are several devices (like serveral HSCDTD008A) , they are distinguishable by the file handle.
This is somewhat incompatible with the current implementation/with arduino approach and  I am not sure how to resolve this.
For the device, the signature name could be extended with a char* for the device string. The arduino implementation could just hand over "Null" as parameter

For the i2c address, it is more complicated and again, I am not sure how to resolve it cleanly. Maybe, the arduino implementation can store it in a struct, that is given to the open, read, and write functions (in the end, that would be a variant of the posix concept as the file handle is a numerical reference to a struct for a particular device). Another approach would be to give the i2c address as additional parameter to the open function. The platform specific implementation could either store it in a struct, a global, or as attribute of an object...